app/controllers/departure/drafts/DashboardController.scala

```diff
@@ -16,37 +16,72 @@
 
 package controllers.departure.drafts
 
-import config.FrontendAppConfig
+import config.{FrontendAppConfig, PaginationAppConfig, SearchResultsAppConfig}
 import controllers.actions._
+import forms.SearchFormProvider
+import models.requests.IdentifierRequest
+import play.api.data.Form
 import play.api.i18n.{I18nSupport, MessagesApi}
-import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import play.twirl.api.HtmlFormat
 import services.DraftDepartureService
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
 import viewModels.drafts.AllDraftDeparturesViewModel
 import views.html.departure.drafts.DashboardView
 
 import javax.inject.Inject
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 
 class DashboardController @Inject() (
   override val messagesApi: MessagesApi,
   identify: IdentifierAction,
   val controllerComponents: MessagesControllerComponents,
   draftDepartureService: DraftDepartureService,
-  view: DashboardView,
-  appConfig: FrontendAppConfig
+  formProvider: SearchFormProvider,
+  paginationAppConfig: PaginationAppConfig,
+  view: DashboardView
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport {
 
+  private val form = formProvider()
+
+  private lazy val pageSize = paginationAppConfig.draftDeparturesNumberOfDrafts
+
   def onPageLoad(): Action[AnyContent] = (Action andThen identify).async {
     implicit request =>
-      draftDepartureService.getAll().map {
-        case Some(draft) =>
-          val toViewModel = AllDraftDeparturesViewModel(draft)
-          Ok(view(toViewModel))
-        case None => Redirect(controllers.routes.ErrorController.technicalDifficulties())
-      }
+      buildView(form, pageSize)(Ok(_))
+  }
+
+  def onSubmit: Action[AnyContent] = (Action andThen identify).async {
+    implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => buildView(formWithErrors, pageSize)(BadRequest(_)),
+          lrn => {
+            val fuzzyLrn: Option[String] = Option(lrn).filter(_.trim.nonEmpty)
+            buildView(form, pageSize, fuzzyLrn, isSearch = true)(Ok(_))
+          }
+        )
+  }
+
+  private def buildView(form: Form[String], pageSize: Int, lrn: Option[String] = None, isSearch: Boolean = false)(
```

Comment:
Maybe flags are better avoided (as in a function should do one thing). Something to consider